tempodb/encoding/vparquet/block.go

```diff
@@ -11,22 +11,22 @@ const (
 	DataFileName = "data.parquet"
 )
 
-type backendBlock struct {
+type BackendBlock struct {
 	meta *backend.BlockMeta
 	r    backend.Reader
 
 	openMtx sync.Mutex
 }
 
-var _ common.BackendBlock = (*backendBlock)(nil)
+var _ common.BackendBlock = (*BackendBlock)(nil)
 
-func newBackendBlock(meta *backend.BlockMeta, r backend.Reader) *backendBlock {
-	return &backendBlock{
+func NewBackendBlock(meta *backend.BlockMeta, r backend.Reader) *BackendBlock {
```

Comment:
I'm wondering whether there is an alternative to making this public. Maybe the CLI could have a thin wrapper around `backend.ContextReader` that implements an `io.ReaderAt` that can be passed into [`parquet.OpenFile()`](https://pkg.go.dev/github.com/segmentio/parquet-go#OpenFile)

Reply:
Used `NewBackendReaderAt()` and passed the reader to `parquet.OpenFile()` instead. No need to export `BackendBlock` nor any of its methods. Nice call.